bpf/lib/common.h

```diff
@@ -395,14 +402,17 @@ enum {
 #define REASON_LB_NO_BACKEND		6
 #define REASON_LB_REVNAT_UPDATE		7
 #define REASON_LB_REVNAT_STALE		8
+#define REASON_FRAG_PACKET		9
+#define REASON_FRAG_PACKET_UPDATE	10
 
 /* Lookup scope for externalTrafficPolicy=Local */
 #define LB_LOOKUP_SCOPE_EXT	0
 #define LB_LOOKUP_SCOPE_INT	1
 
 /* Cilium metrics direction for dropping/forwarding packet */
+#define METRIC_EGRESS   0
 #define METRIC_INGRESS  1
-#define METRIC_EGRESS   2
+#define METRIC_SERVICE  2
```

Comment:
I don't understand, what direction is service? I don't see any other use of this value in the PR either.

Reply:
Here I'm updating the `METRIC_*` direction values to have a consistent 1:1 mapping with the CT ones, so `METRIC_SERVICE` is used for packets with a `CT_SERVICE` direction.

As Martynas noted:
> in the case of a NodePort svc being accessed from outside and a ClusterIP svc from inside the cluster when bpf_sock is disabled, the LB lookup does a CT lookup with dir=CT_SERVICE.

Reply:
For CT, we create independent entries depending on whether it's a direction or if it's a service. For metrics, we are only trying to observe traffic in a particular direction. I don't think it's helpful to expose CT internal entry type details onto the metrics directionality which is intended for user consumption and interpretation of tx/Rx packets.

I think I missed the point where we were trying to take advantage of having the same numeric values for CT and Metric constants. If we find it useful to converge these constant values for some reason, I'm not against that. At the same time, I think it will make metrics direction more confusing if we assume we can just forward the CT entry type as a direction up through to the user.

Reply:
(sorry, I somehow missed this comment yesterday)

> I think I missed the point where we were trying to take advantage of having the same numeric values for CT and Metric constants, would you mind pointing to it?

It started from here: https://github.com/cilium/cilium/pull/13347#issuecomment-707333299

There are 2 things actually:
* the first  `dir` parameter of `ct_lookup4()` to `ipv4_ct_extract_l4_ports()` (as this last function needs to emit metrics with `metrics_update()`)

Reply:
(sorry, I somehow missed this comment yesterday)

> I think I missed the point where we were trying to take advantage of having the same numeric values for CT and Metric constants, would you mind pointing to it?

It started from here: https://github.com/cilium/cilium/pull/13347#issuecomment-707333299

There are 2 things actually:
* first `ipv4_handle_fragmentation()` needs to know the direction of the packet in order to emit metrics. The source for this is the `dir` parameter of `ct_lookup4()`. The initial problem was that this `dir` variable would be sometimes set to `CT_SERVICE`, so it was not obvious how to map it into `METRIC_INGRESS`/`METRIC_EGRESS` (so I proposed to introduce a new `METRIC_SERVICE`).
* next, since we now have 2 sets of constants, `METRIC_{INGRESS, EGRESS, SERVICE}` and `CT_{INGRESS, EGRESS, SERVICE}`, with inconsistent values (e.g. `METRIC_EGRESS` did not match `CT_EGRESS`) I proposed to converge these constant values (in order to get rid of a helper to convert one direction into another, which should reduce the complexity of the program).

Reply:
I see, and I guess it's difficult to pass the actual direction down from higher up the stack due to the number of layers involved and some of the functions also having many parameters.. Hmm.

No solutions come to mind, not sure there's much we can do. 

:+1: